src/components/textarea/textarea.spec.tsx

```diff
@@ -315,38 +315,76 @@ describe("Textarea", () => {
   });
 
   describe(`when the characterLimit prop is passed`, () => {
-    it.each([2, 3, 4])("renders a character counter", (characterLimit) => {
+    let characterLimit = 4;
+    let hintString =
```

Comment:
**polish**: while moving these two variables outside of the three tests on lines 345, 353 and 371 follows DRY principles, it does make the tests a little bit harder to read.